src/Export/DynamicProductGroupService.php

```diff
@@ -76,37 +74,42 @@ class DynamicProductGroupService
 
     private function __construct(
         ContainerInterface $container,
-        CacheItemPoolInterface $cache,
+        CacheHandler $cacheHandler,
```

Comment:
There are many backwards compatible breaks within your changes, which means we would need a major release. Is there any way to reduce the manual effort when upgrading the plugin and having an extension that explicitly overrides this or other classes, which received backwards compatibility breaks?

Reply:
will have to check the possibilities.